subprojects/core/src/main/java/org/gradle/internal/problems/DefaultProblemLocationAnalyzer.java

```diff
@@ -146,6 +146,6 @@ private Location tryGetLocation(StackTraceElement frame) {
             return null;
         }
 
-        return new Location(source.getLongDisplayName(), source.getShortDisplayName(), frame.getFileName(), lineNumber);
+        return new Location(source.getLongDisplayName(), source.getShortDisplayName(), source.getFileName(), lineNumber);
```

Comment:
@donat What is better here? Obtaining the file name from the source or from the stack frame? I'd expect the one from the source is better.

Reply:
Indeed, `source.getFileName()` is better.